bee-node/src/config.rs

```diff
@@ -29,7 +41,18 @@ pub(crate) struct NodeConfigBuilder {
 }
 
 impl NodeConfigBuilder {
-    pub(crate) fn finish(self) -> NodeConfig {
+    /// Creates a node config builder from the local config file.
+    pub fn from_config() -> Result<Self, Error> {
+        match fs::read_to_string(CONFIG_PATH) {
+            Ok(toml) => match toml::from_str::<Self>(&toml) {
```

Comment:
We can probably use `map_err` instead of this match, would be cleaner.

Reply:
sounds like a very good idea. Doing that.

Reply:
it turns out this is actually a bit of a pain, and the code doesn't look any better tbh. Because to be consistent we should use `map` and `map_err` also on the inner match. It then looks like this:
```rust
fs::read_to_string(path)
            .map(|toml| {
                toml::from_str::<Self>(&toml)
                    .map(|config_builder| config_builder)
                    .map_err(|e| Error::NodeConfigBuilderCreationFailure(e))
            })
            .map_err(|e| Error::ConfigFileReadFailure((e)))
```
Do you find that cleaner?

Reply:
and it also doesn't work, because I now I have a cascaded result, so I need to flatten the result.

Reply:
You should keep the first match but not the second. It's to avoid that useless `Ok(config_builder) => Ok(config_builder)` but you don't have that in the first. Also you don't need `.map(|config_builder| config_builder)`, this is already handled by `.map_err`.

Reply:
of course, you're right. I was just porting, but not thinking.